pkg/controller/command/didclient/command_test.go

```diff
@@ -112,3 +158,88 @@ type mockDIDClient struct {
 func (m *mockDIDClient) CreateDID(domain string, opts ...didclient.CreateDIDOption) (*did.Doc, error) {
 	return m.createDIDValue, m.createDIDErr
 }
+
+func randomURL() string {
+	return fmt.Sprintf("localhost:%d", mustGetRandomPort(3))
+}
+
+func mustGetRandomPort(n int) int {
+	for ; n > 0; n-- {
+		port, err := getRandomPort()
+		if err != nil {
+			continue
+		}
+
+		return port
+	}
+	panic("cannot acquire the random port")
+}
+
+func getRandomPort() (int, error) {
```

Comment:
fyi: there's a package out there you can use for this called `freeport`

Reply:
It looks good... but the licence in that project will require me to modify/update our licence in edge-agent: https://github.com/phayes/freeport/blob/master/LICENSE.md

How should I handle that?

Reply:
Actually, I think you can disregard my above comment about modifying the license file... it just says that redistributions have to retain the original copyright notice. I think my including the package, I am doing so. Let me know if you think there's any problems here

Reply:
@DRK3 to be a bit more precise, it requires:

- [ ] To retain the copyright notice, conditions and disclaimers when redistributing their work in source code format
- [X] To retain the copyright notice, conditions and disclaimers when redistributing their work in binary format in the documentation and/or other materials provided with the distribution.
- [ ] To refrain from using the name of the copyright holder and the names of its contributors to endorse or promote products derived from this software without specific prior written permission

I've marked the one I believe applies to our case.

That point is standard for all open source licenses... including ours (see section (4) in the Apache 2.0 [license](https://github.com/trustbloc/edge-agent/blob/master/LICENSE))... we should be maintaining a NOTICE file for all dependencies, not just BSD-3 ones....

@troyronda ^^

Reply:
Double actually... by using httpmock per the suggestions below, I'm not going to need to create a random port anymore.

Reply:
@llorllale @troyronda Also the httpmock project uses the MIT licence

Reply:
Just talked to @fqutishat , he was referring to "net/http/httptest" which I can use to start up a new test server. So you can also disregard what I said about the MIT license